dbms/src/Interpreters/ExpressionAnalyzer.h

```diff
@@ -141,6 +141,11 @@ class ExpressionAnalyzer : private boost::noncopyable
     /// Create Set-s that we can from IN section to use the index on them.
     void makeSetsForIndex();
 
+    NamesAndTypesList* getColumns()
```

Comment:
Why do we need pointer? Function never returns nullptr.
`const NamesAndTypesList & getColumns() const` ?
May be `getUsedColumns` is better name.

Reply:
Fixed